common/src/auth/ServiceConnectionStringParser.cs

```diff
@@ -0,0 +1,228 @@
+        protected Regex HostNameRegex = new Regex(@"[a-zA-Z0-9_\-\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+        protected Regex SharedAccessKeyNameRegex = new Regex(@"^[a-zA-Z0-9_\-@\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+        protected Regex SharedAccessKeyRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+        protected Regex SharedAccessSignatureRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+
+        private string _hostName;
+        private string _serviceName;
+
+        protected ServiceConnectionStringParser()
+        {
+        }
+
+        /// <summary>
+        /// Factory for new Connection String object.
+        /// </summary>
+        /// <remarks>
+        /// The connection string contains a set of information that uniquely identify an IoT Service.
+        ///
+        /// A valid connection string shall be in the following format:
+        /// <code>
+        /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessKey=[Key]
+        /// </code>
+        ///
+        /// This object parse the connection string providing the artifacts to the <see cref="ServiceConnectionString"/> object.
+        /// </remarks>
+        /// <param name="serviceConnectionString">the <code>string</code> with the connection string information.</param>
+        /// <returns>A <code>ServiceConnectionStringParser</code> object with the parsed connection string.</returns>
+        public static ServiceConnectionStringParser Create(string serviceConnectionString)
+        {
+            if (string.IsNullOrWhiteSpace(serviceConnectionString))
+            {
+                throw new ArgumentNullException(nameof(serviceConnectionString));
+            }
+
+            var ServiceConnectionStringParser = new ServiceConnectionStringParser();
+            ServiceConnectionStringParser.Parse(serviceConnectionString);
+
+            return ServiceConnectionStringParser;
+        }
+
+        /// <summary>
+        /// The Service Client Hostname
+        /// </summary>
+        public string HostName
+        {
+            get { return _hostName; }
+            set { SetHostName(value); }
+        }
+
+        /// <summary>
+        /// The Service Access Policy Name
+        /// </summary>
+        public string SharedAccessKeyName { get; internal set; }
+
+        /// <summary>
+        /// The Service Shared Access Key for the specified
+        /// access policy
+        /// </summary>
+        public string SharedAccessKey { get; internal set; }
+
+        /// <summary>
+        /// The Service Shared Access Signature
+        /// </summary>
+        public string SharedAccessSignatureString { get; internal set; }
+
+        /// <summary>
+        /// The Service Name
+        /// </summary>
+        public string ServiceName
```

Comment:
Why does this need a backing field?

Reply:
Do you think of scenario where this set is required elsewhere?

Reply:
This is taken care of. Thanks